File: vector-arithmetic/Fortran/src/gaxpy3.f
        SUBROUTINE gaxpy (a, x, dx, y, dy, n)
                ! Matrix A times vector x plus vector y - row version
                ! using a call to a level one operation
                ! GVL Algorithm 1.1.3

                ! Input:        A, an n x n (square) matrix stored as
                !                       a vector with n^2 entries
                !               x,y real vectors of length n
                ! Output:       Ax + y, stored in vector y

                IMPLICIT none
                DOUBLE PRECISION a(*), x(*), y(*)
                DOUBLE PRECISION dtprod
                INTEGER dx, dy, n

                ! Internal variables
                INTEGER i                  ! index variables
                INTEGER aoff, yoff        ! offset for vectors

                yoff = 1
                DO i = 1,n
                        aoff = (i - 1) * n + 1
                        y(yoff) = y(yoff) + dtprod(a(aoff),1,x,dx,n)
                        yoff = yoff + dy
                END DO
                RETURN
        END SUBROUTINE

File: vector-arithmetic/Matlab/src/gaxpy3.m
function z = gaxpy3( A, x, y )
% Matrix A times vector x plus vector y - row version
% written with colon notation
% GVL Algorithm 1.1.4 update

% Input:    A, an n x n (square) matrix
%           x,y real vectors of length n
% Output:   z = Ax + y, a vector length n

n = size(A);
n = n(1);

z = y;
for i = 1:n
    z(i) = z(i) + A(i,:) * x;
end
end